cpp/benchmarks/common/generate_input.hpp

```diff
@@ -403,3 +403,35 @@ std::unique_ptr<cudf::table> create_random_table(std::vector<cudf::type_id> cons
                                                  row_count num_rows,
                                                  data_profile const& data_params = data_profile{},
                                                  unsigned seed                   = 1);
+
+/**
+ * @brief Generate sequence columns starting with value 0 in first row and increasing by 1 in
+ * subsequent rows.
+ *
+ * If the number of passed types is smaller than the number of requested column, the columns types
+ * with be repeated in round-robin order to fill the table.
+ *
+ * @param dtype_ids Vector of requested column types
+ * @param num_cols Number of columns in the output table
+ * @param num_rows Number of rows in the output table
+ * @param null_probability optional, probability of a null value, <0 implies no null mask.
```

Comment:
Since `null_probability < 0` implies no null mask (unallocated), can you create special cases in the implementation for `null_probability == 0` and `null_probability >= 1` that bypass the random generator and construct a fully valid / fully null mask in the most efficient way?

Reply:
updated.